### PR TITLE
Docker Container Work

### PR DESCRIPTION
**Pull request by da2ce7**

### CODE REVIEW

File: .github/workflows/container.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          context: .
+          load: true
+          push: false
+          tags: index-backend:local
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - id: inspect
+        name: Inspect Inspect
+        run: docker image inspect index-backend:local
```
Comment (josecelano) [inline .github/workflows/container.yml:33]: Hi @da2ce7 when you generate the local docker image here, is that image available when you run:

```
- id: test
  name: Run Integration Tests
  run: ./docker/bin/run-e2e-tests.sh
```

here: https://github.com/torrust/torrust-index-backend/pull/248/files#diff-cf3dd985ec02b2c727885d15dd9a5eca1e2f1ed708cb81406d99f57099aed253R126

I see it's not: https://github.com/torrust/torrust-index-backend/actions/runs/5754190543/job/15598977722?pr=248#step:3:251

You need to save an artefact or push it to a registry.

File: .github/workflows/container.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: Container
```
Comment (josecelano) [inline .github/workflows/container.yml:1]: Hi @da2ce7, I would change the name to something like:

`Generate local docker image`

Or any other alternative, but I think `Container` is too generic.

Why do you create an independent workflow if the putout (the local docker image) is only used in the testing workflow?

I think you can simply move these steps to the `integration` job in the `testing` workflow for now.

**Review by josecelano — CHANGES_REQUESTED** [review]

File: compose.yaml [hunk 3]
```diff
@@ -2,16 +2,12 @@ name: torrust
 services:
 
   idx-back:
-    build:
-      context: .
-      args:
-        RUN_AS_USER: appuser
-        UID: ${TORRUST_IDX_BACK_USER_UID:-1000}
-      target: development
+    image: index-backend:local
```
[the remark this answers is not in the document]
Comment (da2ce7) [reply compose.yaml]: This is not a problem. You can run `docker build --tag index-backend:local .` and it will be included in your local cache.

**Comment by da2ce7** [comment]
closed in preference for:
 - #338

**Pull request closed without merging**